awx/main/tasks/callback.py

```diff
@@ -15,11 +19,67 @@
 from awx.main.utils.update_model import update_model
 from awx.main.queue import CallbackQueueDispatcher
 
+from flags.state import flag_enabled
+
 logger = logging.getLogger('awx.main.tasks.callback')
 
 
+def collect_queries(query_file_contents) -> dict:
+    """
+    collect_queries extracts host queries from the contents of
+    ansible_data.json
+    """
+    result = {}
+
+    try:
+        installed_collections = query_file_contents['installed_collections']
+    except KeyError:
+        logger.error("installed_collections missing in callback response")
+        return result
+
+    for key, value in installed_collections.items():
+        if 'host_query' in value and 'version' in value:
+            result[key] = value
+
+    return result
+
+
+COLLECTION_FILENAME = "ansible_data.json"
+
+
+def try_load_query_file(artifact_dir) -> (bool, dict):
```

Comment:
@pb82 I don't know if this type hint works. Maybe on some versions but not others?

![Screenshot from 2025-01-28 16-04-57](https://github.com/user-attachments/assets/02612a58-4db7-4d17-a346-7b1d4f89cc3d)

Reply:
I'm on Python 3.13 - probably best if I remove it to not cause issues with older versions.

Reply:
You don't have to remove it, there's another way to type hint this same thing that works on more versions.